src/engine/ChunkSystemTileMap.ts

```diff
@@ -0,0 +1,466 @@
+import { Actor } from './Actor';
+import { Vector } from './Algebra';
+import { Class } from './Class';
+import { BoundingBox } from './Collision/BoundingBox';
+import { Configurable } from './Configurable';
+import { Engine } from './Engine';
+import * as Events from './Events';
+import { SpriteSheet } from './Drawing/SpriteSheet';
+import { Cell, TileMap } from './TileMap';
+
+export type ChunkGenerator = (chunkColumn: number, chunkRow: number, chunkSystemTileMap: ChunkSystemTileMap, engine: Engine) => TileMap;
```

Comment:
While doable using this trick:

```typescript
type BaseChunkGenerator = (chunk: TileMap, ...args: Parameters<ChunkGenerator>) => TileMap
```

...I would argue against this, since numerical coordinates passed to `ChunkGenerator` and `BaseChunkGenerator` have different names and semantics (chunk-level coordinates vs cell-level coordinates).

That being said, you are the member of this project and I will follow by your lead. What is your stance on this?